source/agent/addons/quic/QuicTransportStream.cc

```diff
@@ -214,6 +222,20 @@ NAN_SETTER(QuicTransportStream::isMediaSetter)
     obj->m_isMedia = value->ToBoolean(Nan::GetCurrentContext()).ToLocalChecked()->Value();
 }
 
+NAN_GETTER(QuicTransportStream::onDataGetter)
+{
+    QuicTransportStream* obj = Nan::ObjectWrap::Unwrap<QuicTransportStream>(info.Holder());
+    info.GetReturnValue().Set(Nan::New(obj->m_onDataCallback));
+}
+
+NAN_SETTER(QuicTransportStream::onDataSetter)
+{
+    QuicTransportStream* obj = Nan::ObjectWrap::Unwrap<QuicTransportStream>(info.Holder());
+    Nan::Persistent<v8::Value> persistentCallback(value);
```

Comment:
Hi,
I look at the document for Nan::Persistent(https://github.com/nodejs/nan/blob/master/doc/persistent.md#nanpersistent)

The destructor description says "The destructor will dispose the Persistent based on the kResetInDestructor". And I see `kResetInDestructor` is version dependable from the same document. Please check that whether you need `Reset `the `persistentCallback` manually after assignment to `m_onDataCallback`.

Reply:
And if need `Reset` `m_onDataCallback` in destructor.

Reply:
I'll do that in dtor. Another thing I'm going to add is checking the type of `value`, if it's null or undefined, `m_onDataCallback` will be reset.